pkg/sdn/plugin/networkpolicy.go

```diff
@@ -364,7 +278,7 @@ func (np *networkPolicyPlugin) selectPods(npns *npNamespace, lsel *metav1.LabelS
 		glog.Errorf("ValidateNetworkPolicy() failure! Invalid PodSelector: %v", err)
 		return ips
 	}
-	for _, pod := range npns.pods {
+	for _, pod := range np.pods {
```

Comment:
Since you're looping over all pods now you need to check pod.Namespace explicitly